src/commonTest/kotlin/fr/acinq/lightning/channel/states/SpliceTestsCommon.kt

```diff
@@ -0,0 +1,146 @@
+        val (bob4, actionsBob4) = bob3.process(ChannelCommand.MessageReceived(actionsAlice4.findOutgoingMessage<TxAddOutput>()))
+        val (alice5, actionsAlice5) = alice4.process(ChannelCommand.MessageReceived(actionsBob4.findOutgoingMessage<TxComplete>()))
+        val (bob5, actionsBob5) = bob4.process(ChannelCommand.MessageReceived(actionsAlice5.findOutgoingMessage<TxComplete>()))
+        val (bob6, actionsBob6) = bob5.process(ChannelCommand.MessageReceived(actionsAlice5.findOutgoingMessage<CommitSig>()))
+        val (alice6, _) = alice5.process(ChannelCommand.MessageReceived(actionsBob5.findOutgoingMessage<CommitSig>()))
+        val (alice7, actionsAlice7) = alice6.process(ChannelCommand.MessageReceived(actionsBob6.findOutgoingMessage<TxSignatures>()))
+        val (bob7, _) = bob6.process(ChannelCommand.MessageReceived(actionsAlice7.findOutgoingMessage<TxSignatures>()))
+
+        runBlocking { assertIs<Command.Splice.Response>(cmd.replyTo.await()) }
+        assertEquals(alice.commitments.active.size + 1, alice7.commitments.active.size)
+        assertEquals(bob.commitments.active.size + 1, bob7.commitments.active.size)
+        assertIs<LNChannel<Normal>>(alice7)
+        assertIs<LNChannel<Normal>>(bob7)
+        return Pair(alice7, bob7)
+    }
+
+    /** Full remote commit resolution from tx detection to channel close */
+    private fun remoteClose(channel1: LNChannel<Closing>, actions1: List<ChannelAction>, commitment: Commitment, remoteCommitTx: Transaction) {
+        assertEquals(commitment.remoteCommit.txid, remoteCommitTx.txid)
+        assertEquals(0, commitment.remoteCommit.spec.htlcs.size, "this helper only supports remote-closing without htlcs")
+
+        assertIs<LNChannel<Closing>>(channel1)
+        assertEquals(1, actions1.findPublishTxs().size)
+        val claimRemoteDelayedOutputTx = actions1.hasPublishTx(ChannelAction.Blockchain.PublishTx.Type.ClaimRemoteDelayedOutputTx)
+        assertEquals(2, actions1.findWatches<WatchConfirmed>().size)
+        val watchConfirmedCommit = actions1.findWatches<WatchConfirmed>().first { it.txId == commitment.remoteCommit.txid }
+        val watchConfirmedClaimRemote = actions1.findWatches<WatchConfirmed>().first { it.txId == claimRemoteDelayedOutputTx.txid }
+        actions1.has<ChannelAction.Storage.StoreState>()
+
+        // remote commit confirmed
+        val (channel2, actions2) = channel1.process(ChannelCommand.WatchReceived(WatchEventConfirmed(channel1.channelId, watchConfirmedCommit.event, channel1.currentBlockHeight, 42, remoteCommitTx)))
+        assertIs<LNChannel<Closing>>(channel2)
+        actions2.has<ChannelAction.Storage.StoreState>()
+
+        // claim main delayed confirmed
+        val (channel3, actionsAlice4) = channel2.process(ChannelCommand.WatchReceived(WatchEventConfirmed(channel1.channelId, watchConfirmedClaimRemote.event, channel2.currentBlockHeight, 43, claimRemoteDelayedOutputTx)))
+        assertIs<LNChannel<Closed>>(channel3)
```

Comment:
Note that because of type erasure, this isn't actually verifying that `channel3` is closed...
We need to either access a property of `Closed` (which will throw at runtime) or add another assertion:

```suggestion
        assertIs<LNChannel<Closed>>(channel3)
        assertIs<Closed>(channel3.state)
```

Reply:
Fixed in eaf84b70.